marklogic-client-api/src/test/java/com/marklogic/client/test/RowManagerTest.java

```diff
@@ -1479,6 +1485,57 @@ public void testBucketGroup() {
     JsonNode actual = row.getContainer("r");
     assertEquals("group unequal", expect, actual);
   }
+
+  @Test
+  public void testFromDocUrisWithWordQuery() {
+    RowManager rowMgr = Common.client.newRowManager();
+    PlanBuilder p = rowMgr.newPlanBuilder();
+    PlanBuilder.ExportablePlan builtPlan = p.fromDocUris(p.cts.wordQuery("trumpet"), "");
+
+    RowSet<RowRecord> recordRowSet = rowMgr.resultRows(builtPlan);
+    Set<String> uriSet = new HashSet<>();
+    uriSet.add("/optic/test/musician4.json");
+    uriSet.add("/optic/test/musician1.json");
+    Iterator<RowRecord> rows = rowMgr.resultRows(builtPlan).iterator();
+    while (rows.hasNext()){
+      String temp = rows.next().get("uri").toString().replaceAll("^\"|\"$", "");
+      Iterator<String> itr = uriSet.iterator();
+      assertTrue(uriSet.contains(temp));
+      uriSet.remove(temp);
+    }
+  }
+
+  @Test
+  public void testFromDocUrisWithDirectoryQuery() throws Exception {
+    RowManager rowMgr = Common.client.newRowManager();
+    PlanBuilder p = rowMgr.newPlanBuilder();
+    PlanBuilder.ExportablePlan builtPlan = p.fromDocUris(p.cts.directoryQuery("/testFromDocUrisWithDirectoryQuery/"), "");
+    DocumentMetadataHandle meta = new DocumentMetadataHandle().withCollections("testcollection");
+    DataMovementManager moveMgr = Common.client.newDataMovementManager();
+    WriteBatcher batcher = moveMgr.newWriteBatcher();
+    batcher.addAs("/testFromDocUrisWithDirectoryQuery/doc1.txt", meta,"This is doc1");
```

Comment:
I think this test would be a little better if at least one of the 4 docs does not belong to the directory that we're going to query on.